Restructure tools classes based on the usecase

praveenkuttappan opened the pull request:
Initial attempt groups all tools based on the uses. I will add an analyzer to enforce cli pattern based on the namespace to keep cli paths to a set of known groups.
for e.g. 
azsdk pipeline <>
azsdk package <>
azsdk engsys <>
azsdk releaseplan <>
azsdk typespec <>
azsdk template <>|

benbp commented:
A couple comments:

- I took the `pipeline` namespacing to indicate it would be tools related to devops pipeline analysis, not tools specific to the spec generation pipelines. I think most people will have a similar misunderstanding.
- The [new-tool.md](https://github.com/Azure/azure-sdk-tools/blob/main/tools/azsdk-cli/docs/new-tool.md#step-by-step-implementation-guide) doc will need to be updated. The linked section has guidance on only using the `Azure.Sdk.Tools.Cli.Tools` which will need to change to guide LLM coding agents on how to select a namespace/directory.
- I think `Template` should be named `Samples` or `Example` as template is more internal team terminology that may not be obvious to others.

praveenkuttappan commented:
> A couple comments:
> 
> * I took the `pipeline` namespacing to indicate it would be tools related to devops pipeline analysis, not tools specific to the spec generation pipelines. I think most people will have a similar misunderstanding.
> * The [new-tool.md](https://github.com/Azure/azure-sdk-tools/blob/main/tools/azsdk-cli/docs/new-tool.md#step-by-step-implementation-guide) doc will need to be updated. The linked section has guidance on only using the `Azure.Sdk.Tools.Cli.Tools` which will need to change to guide LLM coding agents on how to select a namespace/directory.
> * I think `Template` should be named `Samples` or `Example` as template is more internal team terminology that may not be obvious to others.

I have updated PipelineTool to remove the references to SDK gen. It is a generic tool to get pipeline run status and also renamed Template to Example. I will add changes in new-tool.md

richardpark-msft commented:
Not sure if this is a separate task, but the top level commands still are in a bit of disarray (not alpha-ordered, and not grouped nicely like you have in your comment):

```bash
Commands:
  package            Package management and validation commands
  eng                Internal azsdk engineering system commands
  github-labels      GitHub service labels tools
  start              Starts the MCP server (stdio mode)
  log                Log processing commands
  pipeline           Commands to help with DevOps pipeline
  azp                Azure Pipelines Tool
  generators         Commands that generate files
  release-plan       Manage release plans in AzureDevops
  release-readiness  Checks release readiness of a SDK package.
  sdk-release        Run the release pipeline for the package
  spec-tool          TypeSpec project tools for Azure REST API Specs
  spec-pr            Pull request tools
  spec-workflow      Tools to help with the TypeSpec SDK generation.
  validate-typespec  Run typespec validation
  test-results       Analyze test results
  tsp                Tools for initializing TypeSpec projects and converting existing Azure service swagger definitions to TypeSpec projects
  example            Example tool demonstrating framework features
```

richardpark-msft reviewed (approved):
Let's go ahead and merge this so we don't keep running into constant conflicts with all the other development. :)

richardpark-msft commented:
> Member Author

Really, this is about keeping the CLI commands consistent - right now there's a top level 'start' command, but that's not actually a command group, like we have with the others.
